src/luxonis_ml/embeddings/ldf_emb.py

```diff
@@ -0,0 +1,292 @@
+    Args:
+        qdrant_client: Qdrant client instance.
+        qdrant_collection_name: Name of the Qdrant collection.
+        vector_size: Size of the vector embeddings.
+        all_payloads: List of all payloads.
+    Returns:
+        List of new payloads that are not in the Qdrant database.
+    """
+    # Filter out samples that are already in the Qdrant database
+    search_queries = [SearchRequest(
+        vector=[0] * vector_size,  # Dummy vector
+        filter=models.Filter(
+            must=[
+                models.FieldCondition(
+                    key="sample_id",
+                    match=models.MatchText(text=payload["sample_id"])
+                )
+            ]
+        ),
+        limit=1
+    ) for payload in all_payloads]
+
+
+    search_results = qdrant_client.search_batch(
+        collection_name=qdrant_collection_name,
+        requests=search_queries
+    )
+
+    new_payloads = [all_payloads[i] for i, res in enumerate(search_results) if not res]
+
+    return new_payloads
+
+def _filter_new_samples_by_id(qdrant_client, qdrant_collection_name="mnist", all_payloads=[]):
+    """
+    Filter out samples that are already in the Qdrant database based on their instance ID.
+    Args:
+        qdrant_client: Qdrant client instance.
+        qdrant_collection_name: Name of the Qdrant collection.
+        all_payloads: List of all payloads.
+    Returns:
+        List of new payloads that are not in the Qdrant database.
+    """
+    # Filter out samples that are already in the Qdrant database
+    if len(all_payloads) == 0:
+        print("Payloads list is empty!")
+        return all_payloads
+
+    ids = [payload["instance_id"] for payload in all_payloads]
+    search_results = qdrant_client.retrieve(
+        collection_name=qdrant_collection_name,
+        ids=ids,
+        with_payload=False,
+        with_vectors=False
+    )
+
+    retrieved_ids = [res.id for res in search_results]
+
+    new_payloads = [payload for payload in all_payloads if payload["instance_id"] not in retrieved_ids]
+
+    return new_payloads
+
+def _generate_new_embeddings(ort_session, output_layer_name="/Flatten_output_0", emb_batch_size=64, new_payloads=[], transform=None):
+    """
+    Generate embeddings for new images using a given ONNX runtime session.
+    Args:
+        ort_session: ONNX runtime session.
+        output_layer_name: Name of the output layer in the ONNX model.
+        emb_batch_size: Batch size for generating embeddings.
+        new_payloads: List of new payloads.
+        transform: Optional torchvision transform for preprocessing images.
+    Returns:
+        List of generated embeddings.
+    """
+    # Generate embeddings for the new images using batching
+    new_embeddings = []
+
+    if transform is None:
+        # Define a transformation for resizing and normalizing the images
+        transform = transforms.Compose([
```

Comment:
Does it make sense setting this by default? What is user does want a None transform (just skip it)?

Reply:
I think I programmed it like that because the default model ResNet50 needs an input size of 224x224.
Or in more general case I believe the transform should never be None, it is recommended to atleast resize all the images to the correct size.